### PR TITLE
Relax dynamic filter left expression requirements

### PR DESCRIPTION
**Pull request by lukasz-stec**
Allow dynamic filter sub expression,
referencing the probe side to be any expression
and not only symbol reference.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java [hunk 1]
```diff
@@ -1173,16 +1174,31 @@ public static GroupingSetDescriptor singleGroupingSet(List<String> groupingKeys)
...
+                    rewriteIdentifiersToSymbolReferences(new SqlParser().createExpression(probeExpression, new ParsingOptions())),
+                    operator,
+                    buildAlias,
+                    nullAllowed);
+        }
+
+        public DynamicFilterPattern(Expression probe, ComparisonExpression.Operator operator, String buildAlias, boolean nullAllowed)
         {
-            this.probe = new SymbolAlias(requireNonNull(probeAlias, "probeAlias is null"));
+            this.probe = probe;
```
Comment (skrzypo987) [inline core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java:1201]: require non null

**Review by skrzypo987 — COMMENTED** [review]
Skimmed.
My understanding of the topic is slim so I am not going to approve it.

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java [hunk 2]
```diff
@@ -635,14 +635,14 @@ private DynamicFiltersResult createDynamicFilters(
                             })
                             .map(expression -> {
                                 ComparisonExpression comparison = expression.getComparison();
-                                Symbol leftSymbol = Symbol.from(comparison.getLeft());
-                                Symbol rightSymbol = Symbol.from(comparison.getRight());
-                                boolean alignedComparison = node.getLeft().getOutputSymbols().contains(leftSymbol);
+                                Expression leftExpression = comparison.getLeft();
+                                Expression rightExpression = comparison.getRight();
+                                boolean alignedComparison = isComparisonAligned(node, leftExpression, rightExpression);
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java:640]: you can simplify it: `node.getLeft().getOutputSymbols(extractUnique(leftExpression))`

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java [hunk 3]
```diff
@@ -672,17 +672,23 @@ private DynamicFiltersResult createDynamicFilters(
                     .stream()
                     .map(clause -> {
                         ComparisonExpression comparison = clause.getComparison();
-                        Symbol probeSymbol = Symbol.from(comparison.getLeft());
+                        Expression probeExpression = comparison.getLeft();
                         Symbol buildSymbol = Symbol.from(comparison.getRight());
-                        Type type = symbolAllocator.getTypes().get(probeSymbol);
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java:677]: you could just take type from `buildSymbol` (+ comment) as comparison expression should have same type on both sides

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java [hunk 4]
```diff
@@ -1280,7 +1286,19 @@ private boolean joinDynamicFilteringExpression(Expression expression, Collection
...
             }
-            return comparison.getLeft() instanceof SymbolReference && comparison.getRight() instanceof SymbolReference;
+
+            return dynamicFilteringApplicable(comparison, leftSymbols, rightSymbols);
+        }
+
+        // sub expression that uses right source has to be a symbol (left can be an arbitrary expression)
+        private boolean dynamicFilteringApplicable(ComparisonExpression comparison, Collection<Symbol> leftSymbols, Collection<Symbol> rightSymbols)
+        {
+            if (comparison.getRight() instanceof SymbolReference && rightSymbols.contains(Symbol.from(comparison.getRight()))) {
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java:1296]: make it single expressions:
```
return (comparison.getRight() instanceof SymbolReference && rightSymbols.contains(Symbol.from(comparison.getRight())))
  || (comparison.getLeft() instanceof SymbolReference && rightSymbols.contains(Symbol.from(comparison.getLeft()))
```

`joinComparisonExpression` is making sure that side of expressions are using symbols from both sides of join

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 5]
```diff
@@ -118,6 +119,24 @@ public void testRightEquiJoin()
                                                 tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey")))))));
     }
 
+    @Test
+    public void testRightEquiJoinWIthLeftExpression()
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:123]: type in test name

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 6]
```diff
@@ -118,6 +119,24 @@ public void testRightEquiJoin()
...
+    {
+        assertPlan("SELECT o.orderkey FROM orders o RIGHT JOIN lineitem l ON l.orderkey + 1 = o.orderkey",
+                output(
+                        join(
+                                RIGHT,
+                                ImmutableList.of(equiJoinClause("ORDERS_OK", "expr")),
+                                ImmutableMap.of("ORDERS_OK", "expr"),
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:130]: that dynamic filter should be removed by `RemoveUnsupportedDynamicFilters`. Why it wasn't removed?

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 7]
```diff
@@ -212,17 +247,22 @@ public void testCrossJoinBetweenDF()
     }
 
     @Test
-    public void testCrossJoinInequalityNoDFWithCast()
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:215]: keep the old test too

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java [hunk 8]
```diff
@@ -1173,16 +1174,31 @@ public static GroupingSetDescriptor singleGroupingSet(List<String> groupingKeys)
         return new GroupingSetDescriptor(groupingKeys, 1, globalGroupingSets);
     }
 
+    public static SymbolMapper symbolMapper(SymbolAliases symbolAliases)
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java:1177]: make it private static method of `DynamicFilterPattern`

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java [hunk 9]
```diff
@@ -1173,16 +1174,31 @@ public static GroupingSetDescriptor singleGroupingSet(List<String> groupingKeys)
...
+    {
+        return new SymbolMapper(symbol ->
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java:1179]: Previously missing alias would fail with `missing expression for alias %s`. Preserve this behaviour

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java (hunk 4 above)
Comment (lukasz-stec) [reply core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java]: done (but the second condition uses leftSymbols)

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 5 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java]: fixed

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 7 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:215]: the old test was split into 2 test cases. I can rejoin them into one but there is no point in keeping both.

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java (hunk 8 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java]: done

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java (hunk 9 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java]: done

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 6 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:130]: why should it be removed?
dynamic filter looks like this:` "orderkey" = #df_260` so the expression is a symbol reference

full  plan:
```
Output[orderkey]
│   Layout: [orderkey:bigint]
└─ RightJoin[("orderkey" = "expr")][$hashvalue, $hashvalue_3]
   │   Layout: [orderkey:bigint]
   │   dynamicFilterAssignments = {expr -> #df_260}
   ├─ ScanFilterProject[table = local:orders:sf0.01, filterPredicate = true, dynamicFilters = {"orderkey" = #df_260}]
   │      Layout: [orderkey:bigint, $hashvalue:bigint]
   │      $hashvalue := combine_hash(bigint '0', COALESCE("$operator$hash_code"("orderkey"), 0))
   │      orderkey := tpch:orderkey
   │      tpch:orderstatus
   │          :: [[F], [O], [P]]
   └─ LocalExchange[SINGLE] ()
      │   Layout: [expr:bigint, $hashvalue_3:bigint]
      └─ Project[]
         │   Layout: [expr:bigint, $hashvalue_4:bigint]
         │   $hashvalue_4 := combine_hash(bigint '0', COALESCE("$operator$hash_code"("expr"), 0))
         └─ ScanProject[table = local:lineitem:sf0.01]
                Layout: [expr:bigint]
                expr := ("orderkey_1" + BIGINT '1')
                orderkey_1 := tpch:orderkey

```

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java [hunk 10]
```diff
@@ -1280,7 +1281,18 @@ private boolean joinDynamicFilteringExpression(Expression expression, Collection
                 }
                 comparison = (ComparisonExpression) expression;
             }
-            return comparison.getLeft() instanceof SymbolReference && comparison.getRight() instanceof SymbolReference;
+
+            return dynamicFilteringApplicable(comparison, leftSymbols, rightSymbols);
+        }
+
+        // sub expression that uses right source has to be a symbol (left can be an arbitrary expression)
+        private boolean dynamicFilteringApplicable(ComparisonExpression comparison, Collection<Symbol> leftSymbols, Collection<Symbol> rightSymbols)
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java:1289]: that name is not meaningful. Either inline this method or find better name for it (maybe, `isBuildSideExpressionSymbolReference`)

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java [hunk 11]
```diff
@@ -1280,7 +1281,18 @@ private boolean joinDynamicFilteringExpression(Expression expression, Collection
...
+        {
+            // right is a symbol and there is NO MISALIGNMENT between comparison order and left/right source order
+            // or
+            // left is a symbol and there is MISALIGNMENT between comparison order and left/right source order so the left expression references right source
+            return (comparison.getRight() instanceof SymbolReference && rightSymbols.contains(Symbol.from(comparison.getRight())))
+                    || (comparison.getLeft() instanceof SymbolReference && !leftSymbols.contains(Symbol.from(comparison.getLeft())));
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java:1295]: don't use negation `!leftSymbols`. Rather use `rightSymbols.contains(Symbol.from(comparison.getLeft()))`

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java [hunk 12]
```diff
@@ -1280,7 +1281,18 @@ private boolean joinDynamicFilteringExpression(Expression expression, Collection
...
             }
-            return comparison.getLeft() instanceof SymbolReference && comparison.getRight() instanceof SymbolReference;
+
+            return dynamicFilteringApplicable(comparison, leftSymbols, rightSymbols);
+        }
+
+        // sub expression that uses right source has to be a symbol (left can be an arbitrary expression)
+        private boolean dynamicFilteringApplicable(ComparisonExpression comparison, Collection<Symbol> leftSymbols, Collection<Symbol> rightSymbols)
+        {
+            // right is a symbol and there is NO MISALIGNMENT between comparison order and left/right source order
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java:1291]: Rather than this comment, change comment to something like:
```
Build side expression must be a symbol reference, since DynamicFilterSourceOperator can only collect column values (not expressions)
```

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 6 above)
Comment (sopel39) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:130]: see `io.trino.sql.planner.iterative.rule.RemoveUnsupportedDynamicFilters.Rewriter#isSupportedDynamicFilterExpression`
On probe side, we only support `CAST` and `symbol references`. Hence `LINEITEM_OK + BIGINT '1'` is not supported

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 7 above)
Comment (sopel39) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:215]: which cases are that?

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 6 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:130]: If I understand the above plan correctly, then the `lineitem` scan is built, and the `orders` scan is a probe side.
Since `LINEITEM_OK + BIGINT '1' `is pushed to the scan by PredicatePushDown the dynamic filter becomes :
`ORDERS_OK = expr` where ORDERS_OK is probe side and it's a symbol reference (I guess it doesn't matter at this point as `expr` is also a symbol reference).

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java (hunk 10 above)
Comment (lukasz-stec) [reply core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PredicatePushDown.java]: I like the idea of inlining to avoid finding a good name...

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 6 above)
Comment (sopel39) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:130]: yes, build side complex expressions are fine

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 13]
```diff
@@ -118,6 +119,24 @@ public void testRightEquiJoin()
...
+        assertPlan("SELECT o.orderkey FROM orders o RIGHT JOIN lineitem l ON l.orderkey + 1 = o.orderkey",
+                output(
+                        join(
+                                RIGHT,
+                                ImmutableList.of(equiJoinClause("ORDERS_OK", "expr")),
+                                ImmutableMap.of("ORDERS_OK", "expr"),
+                                anyTree(
+                                        tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
+                                exchange(
+                                        project(
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:134]: what is this project (I think it's hash generator)?

Use `anyTree` (instead of `exchange` and `project`) to make the test more resilient

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 14]
```diff
@@ -167,6 +186,22 @@ public void testCrossJoinInequalityDF()
                                         tableScan("lineitem", ImmutableMap.of("L_ORDERKEY", "orderkey")))))));
     }
 
+    @Test
+    public void testCrossJoinInequalityDFConditionReversedOrder()
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:190]: `testCrossJoinInequalityDFWithReversedCondition`

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 15]
```diff
@@ -212,17 +247,22 @@ public void testCrossJoinBetweenDF()
...
-    public void testCrossJoinInequalityNoDFWithCast()
+    public void testCrossJoinInequalityDFWithCastOnTheLeft()
     {
         assertPlan("SELECT o.comment, l.comment FROM lineitem l, orders o WHERE o.comment < l.comment",
                 anyTree(filter("CAST(L_COMMENT AS varchar(79)) > O_COMMENT",
                         join(
                                 INNER,
                                 ImmutableList.of(),
-                                tableScan("lineitem", ImmutableMap.of("L_COMMENT", "comment")),
+                                filter(TRUE_LITERAL,
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:257]: assert that dynamic filter is now present (that is the other `join` override)

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 16]
```diff
@@ -249,6 +289,22 @@ public void testJoin()
                                                 tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey")))))));
     }
 
+    @Test
+    public void testJoinConditionOrderReversed()
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:293]: `testInnerJoinWithConditionReversed`

Also, isn't there really a test here that has condition reversed for inner join?

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java [hunk 17]
```diff
@@ -1175,14 +1176,23 @@ public static GroupingSetDescriptor singleGroupingSet(List<String> groupingKeys)
...
         private final ComparisonExpression.Operator operator;
         private final SymbolAlias build;
         private final boolean nullAllowed;
 
-        public DynamicFilterPattern(String probeAlias, ComparisonExpression.Operator operator, String buildAlias, boolean nullAllowed)
+        public DynamicFilterPattern(String probeExpression, ComparisonExpression.Operator operator, String buildAlias, boolean nullAllowed)
         {
-            this.probe = new SymbolAlias(requireNonNull(probeAlias, "probeAlias is null"));
+            this(
+                    rewriteIdentifiersToSymbolReferences(new SqlParser().createExpression(probeExpression, new ParsingOptions())),
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java:1187]: nit: it seems that this pattern is used in couple of places, we could have a dedicated utility method for it

**Review by sopel39 — COMMENTED** [review]
lgtm % small comments

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 15 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:269]: done.
I had to manually create the case expression because SqlParser produces a different expression than what is used in a plan (typeOnly flag differs).

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java (hunk 17 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java]: actually, there is already a method PlanBuilder.expression (maybe we should move to ExpressionTestUtils?).
I used it here + refactored other places.

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 16 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java]: I don't see it in this class but I suspect it's tested somehow.
I do feel like having this simple case explicitly tested makes sense but it's actually not related to this change so I can remove it.

**Review by lukasz-stec — COMMENTED** [review]
Addressed the comments + commits squashed

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java (hunk 13 above)
Comment (sopel39) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:140]: > Yup, it's hash. anyTree used.

I don't think you pushed that

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 18]
```diff
@@ -212,17 +253,41 @@ public void testCrossJoinBetweenDF()
...
+
+    private DataType varchar(int size)
+    {
+        return new GenericDataType(
+                Optional.empty(),
+                new Identifier("varchar"),
+                ImmutableList.of(new NumericParameter(Optional.empty(), String.valueOf(size))));
+    }
+
+    private Expression cast(String symbol, DataType asDataType)
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:283]: `cast` -> `typeOnlyCast`

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 19]
```diff
@@ -167,6 +192,22 @@ public void testCrossJoinInequalityDF()
                                         tableScan("lineitem", ImmutableMap.of("L_ORDERKEY", "orderkey")))))));
     }
 
+    @Test
+    public void testCrossJoinInequalityDFWithReversedCondition()
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:196]: `ReversedCondition` -> `ConditionReversed`

File: core/trino-main/src/test/java/io/trino/sql/planner/TestSortExpressionExtractor.java [hunk 20]
```diff
@@ -93,7 +91,7 @@ public void testGetSortExpression()
 
     private Expression expression(String sql)
     {
-        return ExpressionTestUtils.planExpression(metadata, TEST_SESSION, TYPE_PROVIDER, rewriteIdentifiersToSymbolReferences(new SqlParser().createExpression(sql, new ParsingOptions())));
+        return ExpressionTestUtils.planExpression(metadata, TEST_SESSION, TYPE_PROVIDER, PlanBuilder.expression(sql));
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestSortExpressionExtractor.java:94]: separate commit

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java [hunk 21]
```diff
@@ -563,7 +562,7 @@ public static PlanMatchPattern join(
                 new JoinMatcher(
                         joinType,
                         expectedEquiCriteria,
-                        expectedFilter.map(predicate -> rewriteIdentifiersToSymbolReferences(new SqlParser().createExpression(predicate, new ParsingOptions()))),
+                        expectedFilter.map(predicate -> PlanBuilder.expression(predicate)),
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java:565]: using of `PlanBuilder.expression` in existing places should be separate commit

**Review by sopel39 — COMMENTED** [review]
lgtm % small comments

File: core/trino-main/src/test/java/io/trino/sql/planner/TestSortExpressionExtractor.java (hunk 20 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/TestSortExpressionExtractor.java:94]: done

File: core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java (hunk 21 above)
Comment (lukasz-stec) [reply core/trino-main/src/test/java/io/trino/sql/planner/assertions/PlanMatchPattern.java:565]: done

**Review by lukasz-stec — COMMENTED** [review]
PlanBuilder.expression moved to a separate commit + missing anyTree added

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 22]
```diff
@@ -114,10 +121,28 @@ public void testRightEquiJoin()
                                 anyTree(
                                         tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
                                 exchange(
-                                        project(
+                                        anyTree(
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:124]: this should be separate commit

File: core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java [hunk 23]
```diff
@@ -114,10 +121,28 @@ public void testRightEquiJoin()
...
+        assertPlan("SELECT o.orderkey FROM orders o RIGHT JOIN lineitem l ON l.orderkey + 1 = o.orderkey",
+                output(
+                        join(
+                                RIGHT,
+                                ImmutableList.of(equiJoinClause("ORDERS_OK", "expr")),
+                                ImmutableMap.of("ORDERS_OK", "expr"),
+                                anyTree(
+                                        tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
+                                exchange(
+                                        anyTree(
```
Comment (sopel39) [inline core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java:140]: with `anyTree` you can drop `exchange` matching above too

**Review by lukasz-stec — COMMENTED** [review]
Dropped not needed change in the existing test

**Review by sopel39 — APPROVED** [review]
